src/backend/distributed/commands/trigger.c

```diff
@@ -203,45 +255,605 @@ get_relation_trigger_oid_compat(HeapTuple heapTuple)
+	 * relationNameList retrieved from AlterObjectDependsStmt->RangeVar and
+	 * we call this method after standard process utility. So, for the further
+	 * usages, it is certain that the last element in "object" list will always
+	 * be the name of the trigger in either before or after standard process
+	 * utility.
+	 */
+	Value *triggerNameValue = llast(triggerObjectNameList);
+	return triggerNameValue;
+}
+
+
+/*
+ * PreprocessDropTriggerStmt is called before a DROP TRIGGER command has been
+ * executed by standard process utility. This function errors out for
+ * unsupported commands or creates ddl job for supported DROP TRIGGER commands.
+ * The reason we process drop trigger commands before standard process utility
+ * (unlike the other type of trigger commands) is that we act according to trigger
+ * type in CitusLocalTableTriggerCommandDDLJob but trigger wouldn't exist after
+ * standard process utility.
+ */
+List *
+PreprocessDropTriggerStmt(Node *node, const char *queryString)
+{
+	DropStmt *dropTriggerStmt = castNode(DropStmt, node);
+	Assert(dropTriggerStmt->removeType == OBJECT_TRIGGER);
+
+	RangeVar *relation = GetDropTriggerStmtRelation(dropTriggerStmt);
+
+	bool missingOk = true;
+	Oid relationId = RangeVarGetRelid(relation, DROP_TRIGGER_LOCK_MODE, missingOk);
+
+	if (!OidIsValid(relationId))
+	{
+		/* let standard process utility to error out */
+		return NIL;
+	}
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	ErrorIfUnsupportedDropTriggerCommand(dropTriggerStmt);
+
+	if (IsCitusLocalTable(relationId))
+	{
+		char *triggerName = NULL;
+		ExtractDropStmtTriggerAndRelationName(dropTriggerStmt, &triggerName, NULL);
+		return CitusLocalTableTriggerCommandDDLJob(relationId, triggerName,
+												   queryString);
+	}
+
+	return NIL;
+}
+
+
+/*
+ * ErrorIfUnsupportedDropTriggerCommand errors out for unsupported
+ * "DROP TRIGGER triggerName ON relationName" commands.
+ */
+static void
+ErrorIfUnsupportedDropTriggerCommand(DropStmt *dropTriggerStmt)
+{
+	RangeVar *relation = GetDropTriggerStmtRelation(dropTriggerStmt);
+
+	bool missingOk = false;
+	Oid relationId = RangeVarGetRelid(relation, DROP_TRIGGER_LOCK_MODE, missingOk);
+
+	if (!IsCitusTable(relationId))
 	{
 		return;
 	}
 
-	char *relationName = triggerRelation->relname;
+	EnsureCoordinator();
+	ErrorOutForTriggerCommandIfNotCitusLocalTable(relationId);
+}
+
 
-	Assert(relationName != NULL);
-	ereport(ERROR, (errmsg("cannot create trigger on relation \"%s\" because it "
-						   "is either a distributed table or a reference table",
+/*
+ * ErrorOutForTriggerCommandIfNotCitusLocalTable is an helper function
+ * to error out for unsupported trigger commands depending on the citus table
+ * type.
+ */
+void
+ErrorOutForTriggerCommandIfNotCitusLocalTable(Oid relationId)
+{
+	if (IsCitusLocalTable(relationId))
+	{
+		return;
+	}
+
+	char *relationName = get_rel_name(relationId);
+	ereport(ERROR, (errmsg("cannot execute command on table \"%s\" because "
```

Comment:
What about we remove the `cannot execute command on table .. because` part ? because getting an error already implies that the command was not successful. I think that part doesn't add much value.